### PR TITLE
stop using "github.com/golang/protobuf/protoc-gen-go/generator"

### PR DESCRIPTION
**Pull request by jhump**
Latest version of the protobuf runtime has a big deprecation warning that is printed from its `init` function for the `generator` package. So all users of this module's `plugins` package get spam in their process output :(

### CODE REVIEW

File: plugins/names.go [hunk 1]
```diff
@@ -350,6 +349,8 @@ func (n *GoNames) computeTypeOfFieldLocked(fld *desc.FieldDescriptor) {
 		t = gopoet.PointerType(n.GoTypeForMessage(fld.GetMessageType()))
 	case dpb.FieldDescriptorProto_TYPE_ENUM:
 		t = n.GoTypeForEnum(fld.GetEnumType())
+	default:
+		panic(fmt.Sprintf("unrecognized type: %v", fld.GetType()))
```
Comment (jhump) [inline plugins/names.go:353]: This is an unrelated change that addresses a vet warning.

File: plugins/names.go [hunk 2]
```diff
@@ -476,7 +477,63 @@ func (n *GoNames) GoTypeOfResponse(md *desc.MethodDescriptor) gopoet.TypeName {
 // upper-case. If the given symbol starts with an underscore, the underscore is
 // replaced with a capital "X".
 func CamelCase(s string) string {
-	return generator.CamelCase(s)
+	// NB(jh): This is forked from generator.CamelCase in the protobuf runtime.
+	// It is forked to avoid deprecation warnings. That entire package is now
+	// deprecated and even prints warnings in its init function(!!). But its
+	// replacement ("protogen" in the google.golang.org/protobuf module) does
+	// not have an analog for this function.
```
Comment (dragonsinth) [inline plugins/names.go:484]: bastards

File: plugins/names.go [hunk 3]
```diff
@@ -476,7 +477,63 @@ func (n *GoNames) GoTypeOfResponse(md *desc.MethodDescriptor) gopoet.TypeName {
...
+			i++
+			t = append(t, s[i])
+		}
+	}
+	return string(t)
+}
+
+// camelCaseSlice is like CamelCase, but the argument is a slice of strings to
+// be joined with "_".
+func camelCaseSlice(elem []string) string {
```
Comment (dragonsinth) [inline plugins/names.go:525]: Should this be public so that plugin writers can use?

Comment (jhump) [reply plugins/names.go:525]: I tried to put most everything a plugin writer actually needs, with more concrete/semantic method names, in the `Names` type.

Also, this is so simple, I maybe should have just inlined it.
